docs/guidelines/gh-actions-runners.md

```diff
@@ -0,0 +1,65 @@
+            - "Allow actions created by GitHub" can be checked.
+            - Add the specific actions you are going to use into the "Allow specified actions and reusable workflows" section
+        - "Fork pull request workflows from outside collaborators"
+            - The default should be "Require approval for first-time contributors".
+  - All external actions that you plan to use should be examined/audited. This could include examining the repos and using services like [StepSecurity's Action Advisor](https://app.stepsecurity.io/action-advisor).
+- Pin the versions of actions that you are using (use SHA).
+- Use OpenID Connect for access to Azure Resources.
+- Manage secrets securely; they should only be available to workflows and branches where needed.
+    - If available, consider using managed identity/OIDC and Azure Key Vaults for secrets.
+- Limit the permissions/scope of all tokens/credentials, especially the GITHUB_TOKEN.
+- Only refer to values that you control in an action (think risks related to non-validated input).
+- Use branch protection for your repository (workflows should be treated as "sensitive", use branch protection and careful review for any changes).
+  
+
+## Using Self-Hosted Runners
+
+When using GitHub Actions with **self-hosted runners**, it is good practice to:
+
+- Investigate and familiarize yourself with the [Hosting your own runners](https://docs.github.com/en/actions/hosting-your-own-runners) documentation.
+- Investigate and familiarize yourself with the [Hardening for self-hosted runners](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#hardening-for-self-hosted-runners).
+- **NEVER** use with **public** repos.
+- Involve the team, create, and keep up to date a **threat model** of your self-hosted runner setup. The threat model should at least include:
+    - A data flow/architecture diagram including key components and code/information flows.
+    - A list of identified threats/risks.
+    - Information on how identified threats/risks are managed.
+    - The threat model should be accepted by product owners/risk owners.
+- For the hosting environment:
+    - We prefer ephemeral environments, non-persistent environments.
+    - [Just-in-time runners](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-just-in-time-runners) may be an option.
+    - Use dedicated hosts/machines.
+    - Only do one job at a time on the runner (don't do auto-scaling with persistent runners, use the --ephemeral flag).
+    - Follow hardening guidelines for the hosting platform (if they exist).
+    - The environment should be isolated by default.
+    - Network egress/ingress should be managed (firewall, NSG rules, ++).
+    - The runner code should execute with the least possible privilege level.
+    - Environments should be refreshed regularly (update, patch) (weekly).
+    - Each job should do a proper clean-up.
```

Comment:
I feel this is very important if the runner is not ephemeral, and a suggestion could be to have it in the bottom and in bold or similar to emphasize it

Reply:
I move it to the bottom of this section and make it bold